spyder/plugins/ipythonconsole/utils/kernelspec.py

```diff
@@ -113,10 +115,18 @@ def __init__(self, is_cython=False, is_pylab=False,
     def argv(self):
         """Command to start kernels"""
         # Python interpreter used to start kernels
+        if self.is_environment:
+            self.set_conf(
+                'executable', self.path_to_environment,
+                section='main_interpreter')
+            self.set_conf('default', False, section='main_interpreter')
+            self.set_conf('custom', True, section='main_interpreter')
```

Comment:
I think here we need just to put `pyexec =  self.path_to_environment` (or the revised version of the variable name `self.path_to_custom_interpreter`)?

Reply:
Or even just remove this validation since the new case is being handled at below line 128?